src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -0,0 +1,615 @@
+end
+
+
+function Base.show(io::IO, semi::SemidiscretizationCoupled)
+  @nospecialize semi # reduce precompilation time
+
+  print(io, "SemidiscretizationCoupled($(semi.semis))")
+end
+
+function Base.show(io::IO, ::MIME"text/plain", semi::SemidiscretizationCoupled)
+  @nospecialize semi # reduce precompilation time
+
+  if get(io, :compact, false)
+    show(io, semi)
+  else
+    summary_header(io, "SemidiscretizationCoupled")
+    summary_line(io, "#spatial dimensions", ndims(semi.semis[1]))
+    summary_line(io, "#systems", nsystems(semi))
+    for i in eachsystem(semi)
+      summary_line(io, "system", i)
+      mesh, equations, solver, _ = mesh_equations_solver_cache(semi.semis[i])
+      summary_line(increment_indent(io), "mesh", mesh |> typeof |> nameof)
+      summary_line(increment_indent(io), "equations", equations |> typeof |> nameof)
+      summary_line(increment_indent(io), "initial condition", semi.semis[i].initial_condition)
+      # no boundary conditions since that could be too much
+      summary_line(increment_indent(io), "source terms", semi.semis[i].source_terms)
+      summary_line(increment_indent(io), "solver", solver |> typeof |> nameof)
+    end
+    summary_line(io, "total #DOFs", ndofs(semi))
+    summary_footer(io)
+  end
+end
+
+
+function print_summary_semidiscretization(io::IO, semi::SemidiscretizationCoupled)
+  show(io, MIME"text/plain"(), semi)
+  println(io, "\n")
+  for i in eachsystem(semi)
+    mesh, equations, solver, _ = mesh_equations_solver_cache(semi.semis[i])
+    summary_header(io, "System #$i")
+
+    summary_line(io, "mesh", mesh |> typeof |> nameof)
+    show(increment_indent(io), MIME"text/plain"(), mesh)
+
+    summary_line(io, "equations", equations |> typeof |> nameof)
+    show(increment_indent(io), MIME"text/plain"(), equations)
+
+    summary_line(io, "solver", solver |> typeof |> nameof)
+    show(increment_indent(io), MIME"text/plain"(), solver)
+
+    summary_footer(io)
+    println(io, "\n")
+  end
+end
+
+
+@inline Base.ndims(semi::SemidiscretizationCoupled) = ndims(semi.semis[1])
+
+@inline nsystems(semi::SemidiscretizationCoupled) = length(semi.semis)
+
+@inline eachsystem(semi::SemidiscretizationCoupled) = Base.OneTo(nsystems(semi))
+
+@inline Base.real(semi::SemidiscretizationCoupled) = promote_type(real.(semi.semis)...)
+
+@inline Base.eltype(semi::SemidiscretizationCoupled) = promote_type(eltype.(semi.semis)...)
+
+@inline function ndofs(semi::SemidiscretizationCoupled)
+  sum(ndofs, semi.semis)
+end
+
+@inline function nelements(semi::SemidiscretizationCoupled)
+  return sum(semi.semis) do semi_
+    mesh, equations, solver, cache = mesh_equations_solver_cache(semi_)
+
+    nelements(mesh, solver, cache)
+  end
+end
+
+@inline function total_volume(semi::SemidiscretizationCoupled)
+  sum(semi.semis) do semi_
+    mesh, equations, solver, cache = mesh_equations_solver_cache(semi_)
+    total_volume(mesh, solver, cache)
+  end
+end
```

Comment:
Not covered in tests at the moment

Reply:
Yes, and `total_volume` is non-functional anyways; I removed it in 101a12bc1

Reply:
We'll keep the `nelements` implementation for now.